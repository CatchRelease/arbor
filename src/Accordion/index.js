import styled from '@emotion/styled';

import Box from '../Box';

const Accordion = styled(Box)`
  border: 1px solid ${props => props.theme.colors.border.default};
  border-radius: ${props => props.theme.borderRadius.large};
`;

export default Accordion;
