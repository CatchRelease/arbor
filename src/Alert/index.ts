export { default as BannerAlert } from './BannerAlert';
export { default as CardAlert } from './CardAlert';
export { default as InlineAlert } from './InlineAlert';
export { default as PaneAlert } from './PaneAlert';
export { default as ToastAlert } from './ToastAlert';
