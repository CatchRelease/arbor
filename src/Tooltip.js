import React from 'react';
import { injectGlobal } from 'react-emotion';
import { withTheme } from 'emotion-theming';
import PropTypes from 'prop-types';
import 'react-tippy/dist/tippy.css';
import { Tooltip as TippyTooltip } from 'react-tippy';

import Text from './Text';

// eslint-disable-next-line no-unused-expressions
injectGlobal`
  .tippy-tooltip.arbor-theme {
    padding: 0;
    border-radius: 3px;
  }
`;

export const Tooltip = ({ text, theme, children, ...props }) => {
  const tooltipText = (
    <Text
      px="smaller"
      py="smallest"
      theme={theme}
      fontSize="size2"
      color="white"
    >
      {text}
    </Text>
  );

  return (
    <TippyTooltip
      {...{
        ...props,
        theme: 'arbor',
        arrow: true,
        html: tooltipText,
        distance: 8,
        duration: 300,
        updateDuration: 0,
        arrowSize: 'small',
        animation: 'fade',
        inertia: true,
        animateFill: false
      }}
    >
      {children}
    </TippyTooltip>
  );
};

Tooltip.propTypes = {
  /**
   * Text to display within the tooltip when it is displayed
   * */
  text: PropTypes.string.isRequired,

  /**
   * Node which will trigger the tooltip. This should be either an Icon, Button,
   * or Link.
   * */
  children: PropTypes.element.isRequired
};

export default withTheme(Tooltip);
