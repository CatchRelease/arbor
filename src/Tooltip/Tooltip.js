/** @jsx jsx */
import React from 'react'; /* eslint-disable-line no-unused-vars */
import { Global, css, jsx } from '@emotion/core';
import { ThemeProvider, withTheme } from 'emotion-theming';
import PropTypes from 'prop-types';
import Tippy from '@tippyjs/react';
import 'tippy.js/dist/tippy.css';
import { animateFill, sticky } from 'tippy.js';

import StyledTooltipContent from './StyledTooltipContent';

const globalTippyStyles = (theme) => {
  const arrowColor = theme.colors.monochrome.grey90;

  return css`
    .tippy-popper {
      .tippy-tooltip.arbor-theme {
        background: none;
        padding: 0;
      }
    }

    .tippy-popper[x-placement='bottom'] .tippy-tooltip.arbor-theme [x-arrow] {
      border-bottom-color: ${arrowColor};
    }

    .tippy-popper[x-placement='top'] .tippy-tooltip.arbor-theme [x-arrow] {
      border-top-color: ${arrowColor};
    }

    .tippy-popper[x-placement='left'] .tippy-tooltip.arbor-theme [x-arrow] {
      border-left-color: ${arrowColor};
    }

    .tippy-popper[x-placement='right'] .tippy-tooltip.arbor-theme [x-arrow] {
      border-right-color: ${arrowColor};
    }
  `;
};

const Tooltip = ({ content, children, theme, position, ...props }) => {
  const tooltipContent = (
    <ThemeProvider theme={theme}>
      <StyledTooltipContent>{content}</StyledTooltipContent>
    </ThemeProvider>
  );

  return (
    <>
      <Global styles={globalTippyStyles} />

      <Tippy
        {...{
          theme: 'arbor',
          arrow: true,
          content: tooltipContent,
          duration: 300,
          animation: 'fade',
          inertia: true,
          plugins: [animateFill, sticky],
          placement: position,
          ...props
        }}
      >
        {children}
      </Tippy>
    </>
  );
};

Tooltip.propTypes = {
  /**
   * Node which will trigger the tooltip. This should be either an Icon, Button,
   * or Link.
   * */
  children: PropTypes.element.isRequired,

  /**
   * Content to display within the tooltip when it is displayed
   * */
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,

  /**
   * Position for the tooltip (deprecated, use placement instead)
   * */
  position: PropTypes.string,

  /**
   * Theme used for styling the Tooltip.
   */
  theme: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

Tooltip.defaultProps = {
  position: 'top'
};

export default withTheme(Tooltip);
