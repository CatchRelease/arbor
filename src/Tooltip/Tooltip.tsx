import { ComponentProps, ReactNode, useState, VFC } from 'react';
import { css } from '@emotion/react';
import Tippy from '@tippyjs/react/headless';
import { sticky } from 'tippy.js';
import { motion } from 'framer-motion';

import Box from '../Box';
import Card from '../Card';
import Grid from '../Grid';
import Heading from '../Heading';
import Text from '../Text';

type TippyProps = ComponentProps<typeof Tippy>;

type Props = {
  children: TippyProps['children'];
  content?: ReactNode;
  title?: ReactNode;
};

type TippyRenderAttributes = Parameters<
  Exclude<TippyProps['render'], undefined>
>[0];

const Tooltip: VFC<Props> = ({ children, content, title, ...props }) => {
  const [mounted, setMounted] = useState(false);

  const lazyPlugin = {
    fn: () => ({
      onMount: () => setMounted(true),
      onHide: () => setMounted(false)
    })
  };

  const springConfig = {
    type: 'spring',
    damping: 24,
    stiffness: 480,
    restSpeed: 0.1,
    restDelta: 0.6
  };

  const variants = {
    visible: { opacity: 1, scale: 1 },
    hidden: { opacity: 0, scale: 0.5 }
  };

  const renderTooltip = (attrs: TippyRenderAttributes) => (
    <Box
      as={motion.div}
      initial="hidden"
      variants={variants}
      animate={mounted ? 'visible' : 'hidden'}
      transition={springConfig}
      {...attrs}
    >
      <Card
        bg="monochrome.grey90"
        borderRadius="small"
        boxShadow="elevation1"
        color="monochrome.white"
        gridGap="smallest"
        fontSize="size3"
        maxWidth="200px"
        p="smaller"
        css={css`
          word-break: break-word;
        `}
        {...props}
      >
        {title && (
          <Heading.H3 color="inherit" mb="0">
            {title}
          </Heading.H3>
        )}
        {content && (
          <Text
            as={Grid}
            fontSize="inherit"
            color="inherit"
            textAlign="inherit"
          >
            {content}
          </Text>
        )}
      </Card>
    </Box>
  );

  return (
    <Tippy
      render={(attrs) => (mounted ? renderTooltip(attrs) : '')}
      offset={[0, 4]}
      plugins={[sticky, lazyPlugin]}
      popperOptions={{
        modifiers: [
          {
            name: 'preventOverflow',
            options: {
              padding: 4
            }
          }
        ]
      }}
    >
      {children}
    </Tippy>
  );
};

export default Tooltip;
