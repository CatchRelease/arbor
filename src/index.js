export * from './theme';
export { default as Box } from './Box';
export { default as Button } from './Button';
export { default as Card } from './Card';
export { default as CardActions } from './CardActions';
export { default as CardPreview } from './CardPreview';
export { default as CardRow } from './CardRow';
export { default as Checkbox } from './Checkbox';
export { default as Dropdown } from './Dropdown';
export { default as Flex } from './Flex';
export { default as Grid } from './Grid';
export { default as Heading } from './Heading';
export { default as Icon } from './Icon';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Link } from './Link';
export { default as Menu } from './Menu';
export { default as Pagination } from './Pagination';
export { default as Paragraph } from './Paragraph';
export { default as Popover } from './Popover';
export { default as Text } from './Text';
export { default as Tooltip } from './Tooltip';

export { default as reset } from './reset';
export { default as theme } from './theme';
