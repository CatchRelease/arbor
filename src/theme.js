import { rem } from 'polished';

export const breakpoints = ['768px'];

export const blues = {
  blueLighter: '#6688FF',
  blueLight: '#5577FF',
  blue: '#4466EE',
  blueDark: '#4455CC',
  blueDarkest: '#2E374E'
};

export const greys = {
  grey10: '#EEEEEE',
  grey20: '#DDDDDD',
  grey30: '#CCCCCC',
  grey40: '#BBBBBB',
  grey50: '#AAAAAA',
  grey60: '#999999',
  grey70: '#888888',
  grey80: '#666666',
  grey90: '#444444',
  grey100: '#222222'
};

export const whites = {
  white: '#FFFFFF',
  white10: '#FAFAFA'
};

export const reds = {
  redLighter: '#FF6655',
  redLight: '#FF4444',
  red: '#EE2200',
  redDark: '#DD1100'
};

export const greens = {
  greenLighter: '#00CC66',
  greenLight: '#00BB44',
  green: '#00AA44',
  greenDark: '#008833'
};

export const black = '#000000';

export const colors = {
  black,
  ...blues,
  ...greys,
  ...whites,
  ...greens,
  ...reds
};

export const fontSizes = {
  size1: rem('12px'),
  size2: rem('14px'),
  size3: rem('16px'),
  size4: rem('18px'),
  size5: rem('22px'),
  size6: rem('28px'),
  size7: rem('34px')
};

export const fontWeights = {
  regular: 400,
  medium: 600,
  bold: 700
};

export const lineHeights = {
  small: 1.25,
  large: 1.5
};

export const typography = {
  text: {
    ui: {
      fontSize: fontSizes.size2,
      color: colors.grey100
    },
    longForm: {
      fontSize: fontSizes.size3,
      color: colors.grey100
    },
    tiny: {
      fontSize: fontSizes.size1,
      color: colors.grey80
    }
  }
};

export const brandFont = 'Motiva Sans';

export const spacings = {
  smallest: rem('4px'),
  smaller: rem('8px'),
  small: rem('12px'),
  regular: rem('16px'),
  large: rem('24px'),
  larger: rem('32px'),
  largest: rem('64px')
};

export const borderRadius = {
  small: '3px',
  large: '6px'
};

export const borderWidth = {
  small: '1px',
  medium: '2px',
  large: '4px'
};

export default {
  borderRadius,
  borderWidth,
  brandFont,
  breakpoints,
  colors,
  fontWeights,
  fontSizes,
  iconFontPrefix: 'fa',
  lineHeights,
  space: spacings,
  typography
};
