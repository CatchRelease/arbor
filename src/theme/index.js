import { rem, transparentize } from 'polished';
import { intent, monochrome, palette, primary, secondary } from './colors';

export const breakpointsMap = {
  sm: '512px',
  md: '768px',
  lg: '1024px',
  xl: '1600px'
};

export const breakpoints = Object.values(breakpointsMap);

export const blues = {
  blueLighter: '#6688FF',
  blueLight: '#5577FF',
  blue: '#4466EE',
  blueDark: '#4455CC',
  blueDarkest: '#2E374E'
};

export const greys = {
  grey10: '#F8F9F9',
  grey20: '#F2F2F2',
  grey30: '#EBECEC',
  grey40: '#E0E1E2',
  grey50: '#BEBEC0',
  grey60: '#989A9C',
  grey70: '#78797D',
  grey80: '#5B5D61',
  grey90: '#25272D',
  grey100: '#0F0F10'
};

export const whites = {
  white: '#FFFFFF'
};

export const reds = {
  redLighter: '#FF6655',
  redLight: '#FF4444',
  red: '#EE2200',
  redDark: '#DD1100'
};

export const greens = {
  greenLighter: '#00CC66',
  greenLight: '#00BB44',
  green: '#00AA44',
  greenDark: '#008833'
};

export const black = '#000000';

export const bronzes = {
  bronzeLight: '#CC9900',
  bronze: '#CC7700',
  bronzeDark: '#AA5500'
};

export const backgroundColors = {
  muted: greys.grey10,
  default: greys.grey20,
  white: whites.white
};

export const borderColors = {
  muted: greys.grey30,
  default: greys.grey40
};

export const brandColors = {
  get light() {
    return transparentize(0.91, this.dark);
  },
  default: '#007558',
  dark: '#00533E',
  darkest: '#001C15'
};

export const iconColors = {
  disabled: transparentize(0.3, greys.grey50),
  default: greys.grey60
};

export const textColors = {
  disabled: transparentize(0.3, greys.grey70),
  muted: greys.grey80,
  default: greys.grey90,
  dark: greys.grey100
};

export const colors = {
  /* START: deprecated -- Color definition and variables are being revamped */
  black,
  ...blues,
  ...greys,
  ...whites,
  ...greens,
  ...reds,
  ...bronzes,
  background: { ...backgroundColors },
  brand: { ...brandColors },
  border: { ...borderColors },
  icon: { ...iconColors },
  text: { ...textColors },
  /* END: deprecated */
  intent,
  monochrome,
  palette,
  primary,
  secondary
};

export const brandFont = 'Motiva Sans';

export const fonts = {
  default: brandFont,
  icon: 'Arbor'
};

const sizes = {
  size1: rem('11px'),
  size2: rem('12px'),
  size3: rem('13px'),
  size4: rem('14px'),
  size5: rem('16px'),
  size6: rem('20px'),
  size7: rem('24px'),
  size8: rem('28px'),
  size9: rem('34px')
};

const inputSizes = {
  default: sizes.size4,
  mobile: sizes.size5
};

export const fontSizes = {
  ...sizes,
  input: { ...inputSizes }
};

export const fontWeights = {
  regular: 400,
  medium: 500,
  bold: 700
};

export const icons = {
  annotation: '\f039',
  calendar: '\f035',
  caret: '\f037',
  chatBubbleOutline: '\f003',
  checkboxChecked: '\f030',
  checkboxIndeterminate: '\f02f',
  checkboxMarkedCircleOutline: '\f000',
  chevron: '\f03a',
  crLogo: '\f02e',
  cross: '\f03b',
  download: '\f004',
  swap: '\f038',
  symbolCircle: '\f034',
  thumbDownOutline: '\f001',
  thumbUpOutline: '\f002',
  trash: '\f036'
};

export const lineHeights = {
  small: 1.25,
  regular: 1,
  large: 1.5
};

export const typography = {
  text: {
    ui: {
      fontSize: fontSizes.size4
    },
    longForm: {
      fontSize: fontSizes.size5
    },
    tiny: {
      fontSize: fontSizes.size2
    }
  }
};

export const spacings = {
  smallest: rem('4px'),
  smaller: rem('8px'),
  small: rem('12px'),
  regular: rem('16px'),
  large: rem('24px'),
  larger: rem('32px'),
  largest: rem('64px')
};

export const borderRadius = {
  small: '3px',
  large: '6px',
  larger: '12px'
};

export const borderWidth = {
  small: '1px',
  medium: '2px',
  large: '4px'
};

const borders = {
  default: `${borderWidth.small} solid ${borderColors.default}`
};

const blurryShadowColor = transparentize(0.7, greys.grey90);
const borderShadowColor = transparentize(0.53, greys.grey90);
export const boxShadows = {
  elevation0: `0 0 1px ${borderShadowColor}`,
  elevation1: `0 0 1px ${borderShadowColor}, 0 2px 4px -2px ${blurryShadowColor}`,
  elevation2: `0 0 1px ${borderShadowColor}, 0 5px 8px -4px ${blurryShadowColor}`,
  elevation3: `0 0 1px ${borderShadowColor}, 0 8px 10px -4px ${blurryShadowColor}`,
  elevation4: `0 0 1px ${borderShadowColor}, 0 16px 24px -8px ${blurryShadowColor}`
};

export const avatarSizes = {
  default: '24px'
};

export default {
  avatarSizes,
  backgroundColors,
  borderColors,
  borderWidth,
  borders,
  brandColors,
  brandFont,
  breakpoints,
  breakpointsMap,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  icons,
  iconColors,
  iconFontPrefix: 'ar',
  lineHeights,
  radii: borderRadius,
  shadows: boxShadows,
  space: spacings,
  textColors,
  typography
};
