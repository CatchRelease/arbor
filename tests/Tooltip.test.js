import React from 'react';
import { shallow } from 'enzyme';

import theme from '../src/theme';
import { Tooltip } from '../src/Tooltip';

describe('<Tooltip />', () => {
  it('properly renders a tooltip', () => {
    const tree = shallow(
      <Tooltip theme={theme} text="Hello World">
        <p>I have a tooltip</p>
      </Tooltip>
    );

    expect(tree).toMatchSnapshot();
  });
});
