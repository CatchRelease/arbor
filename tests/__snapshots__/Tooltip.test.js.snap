// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> properly renders a tooltip 1`] = `
<WithTheme(Tooltip)
  text="Hello World"
>
  <Tooltip
    text="Hello World"
    theme={
      Object {
        "borderRadius": Object {
          "large": "6px",
          "small": "3px",
        },
        "borderWidth": Object {
          "large": "4px",
          "medium": "2px",
          "small": "1px",
        },
        "brandFont": "Motiva Sans",
        "breakpoints": Array [
          "512px",
          "768px",
          "1024px",
          "1600px",
        ],
        "colors": Object {
          "black": "#000000",
          "blue": "#4466EE",
          "blueDark": "#4455CC",
          "blueDarkest": "#2E374E",
          "blueLight": "#5577FF",
          "blueLighter": "#6688FF",
          "bronze": "#CC7700",
          "bronzeDark": "#AA5500",
          "bronzeLight": "#CC9900",
          "green": "#00AA44",
          "greenDark": "#008833",
          "greenLight": "#00BB44",
          "greenLighter": "#00CC66",
          "grey10": "#EEEEEE",
          "grey100": "#222222",
          "grey20": "#DDDDDD",
          "grey30": "#CCCCCC",
          "grey40": "#BBBBBB",
          "grey50": "#AAAAAA",
          "grey60": "#999999",
          "grey70": "#888888",
          "grey80": "#666666",
          "grey90": "#444444",
          "red": "#EE2200",
          "redDark": "#DD1100",
          "redLight": "#FF4444",
          "redLighter": "#FF6655",
          "white": "#FFFFFF",
          "white10": "#F5F5F5",
        },
        "fontSizes": Object {
          "size1": "0.75rem",
          "size2": "0.875rem",
          "size3": "1rem",
          "size4": "1.125rem",
          "size5": "1.375rem",
          "size6": "1.75rem",
          "size7": "2.125rem",
        },
        "fontWeights": Object {
          "bold": 700,
          "medium": 600,
          "regular": 400,
        },
        "iconFontPrefix": "fa",
        "lineHeights": Object {
          "large": 1.5,
          "small": 1.25,
        },
        "space": Object {
          "large": "1.5rem",
          "larger": "2rem",
          "largest": "4rem",
          "regular": "1rem",
          "small": "0.75rem",
          "smaller": "0.5rem",
          "smallest": "0.25rem",
        },
        "typography": Object {
          "text": Object {
            "longForm": Object {
              "color": "#222222",
              "fontSize": "1rem",
            },
            "tiny": Object {
              "color": "#666666",
              "fontSize": "0.75rem",
            },
            "ui": Object {
              "color": "#222222",
              "fontSize": "0.875rem",
            },
          },
        },
      }
    }
  >
    <ForwardRef(render)
      styles={
        Object {
          "map": undefined,
          "name": "957k5c-globalTippyStyles",
          "next": undefined,
          "styles": ".tippy-tooltip.arbor-theme{border-radius:3px;padding:0;}label:globalTippyStyles;",
        }
      }
    >
      <InnerGlobal
        cache={
          Object {
            "insert": [Function],
            "inserted": Object {},
            "key": "css",
            "nonce": undefined,
            "registered": Object {},
            "sheet": StyleSheet {
              "before": null,
              "container": <head>
                <style
                  data-emotion="css-global"
                >
                  
                  .tippy-tooltip.arbor-theme{border-radius:3px;padding:0;}
                </style>
              </head>,
              "ctr": 0,
              "isSpeedy": false,
              "key": "css",
              "nonce": undefined,
              "tags": Array [],
            },
          }
        }
        serialized={
          Object {
            "map": undefined,
            "name": "957k5c-globalTippyStyles",
            "next": undefined,
            "styles": ".tippy-tooltip.arbor-theme{border-radius:3px;padding:0;}label:globalTippyStyles;",
          }
        }
      />
    </ForwardRef(render)>
    <Tooltip
      animateFill={false}
      animation="fade"
      arrow={true}
      arrowSize="small"
      className=""
      delay={0}
      disabled={false}
      distance={8}
      duration={300}
      followCursor={false}
      hideDelay={0}
      hideDuration={375}
      hideOnClick={true}
      html={
        <ForwardRef(render)
          color="white"
          fontSize="size2"
          px="smaller"
          py="smallest"
          theme={
            Object {
              "borderRadius": Object {
                "large": "6px",
                "small": "3px",
              },
              "borderWidth": Object {
                "large": "4px",
                "medium": "2px",
                "small": "1px",
              },
              "brandFont": "Motiva Sans",
              "breakpoints": Array [
                "512px",
                "768px",
                "1024px",
                "1600px",
              ],
              "colors": Object {
                "black": "#000000",
                "blue": "#4466EE",
                "blueDark": "#4455CC",
                "blueDarkest": "#2E374E",
                "blueLight": "#5577FF",
                "blueLighter": "#6688FF",
                "bronze": "#CC7700",
                "bronzeDark": "#AA5500",
                "bronzeLight": "#CC9900",
                "green": "#00AA44",
                "greenDark": "#008833",
                "greenLight": "#00BB44",
                "greenLighter": "#00CC66",
                "grey10": "#EEEEEE",
                "grey100": "#222222",
                "grey20": "#DDDDDD",
                "grey30": "#CCCCCC",
                "grey40": "#BBBBBB",
                "grey50": "#AAAAAA",
                "grey60": "#999999",
                "grey70": "#888888",
                "grey80": "#666666",
                "grey90": "#444444",
                "red": "#EE2200",
                "redDark": "#DD1100",
                "redLight": "#FF4444",
                "redLighter": "#FF6655",
                "white": "#FFFFFF",
                "white10": "#F5F5F5",
              },
              "fontSizes": Object {
                "size1": "0.75rem",
                "size2": "0.875rem",
                "size3": "1rem",
                "size4": "1.125rem",
                "size5": "1.375rem",
                "size6": "1.75rem",
                "size7": "2.125rem",
              },
              "fontWeights": Object {
                "bold": 700,
                "medium": 600,
                "regular": 400,
              },
              "iconFontPrefix": "fa",
              "lineHeights": Object {
                "large": 1.5,
                "small": 1.25,
              },
              "space": Object {
                "large": "1.5rem",
                "larger": "2rem",
                "largest": "4rem",
                "regular": "1rem",
                "small": "0.75rem",
                "smaller": "0.5rem",
                "smallest": "0.25rem",
              },
              "typography": Object {
                "text": Object {
                  "longForm": Object {
                    "color": "#222222",
                    "fontSize": "1rem",
                  },
                  "tiny": Object {
                    "color": "#666666",
                    "fontSize": "0.75rem",
                  },
                  "ui": Object {
                    "color": "#222222",
                    "fontSize": "0.875rem",
                  },
                },
              },
            }
          }
        >
          Hello World
        </ForwardRef(render)>
      }
      inertia={true}
      interactive={false}
      interactiveBorder={2}
      multiple={false}
      offset={0}
      onHidden={[Function]}
      onHide={[Function]}
      onRequestClose={[Function]}
      onShow={[Function]}
      onShown={[Function]}
      popperOptions={Object {}}
      position="top"
      size="regular"
      sticky={false}
      stickyDuration={200}
      style={Object {}}
      theme="arbor"
      touchHold={false}
      trigger="mouseenter focus"
      unmountHTMLWhenHide={false}
      updateDuration={0}
    >
      <div
        className=""
        style={
          Object {
            "display": "inline",
          }
        }
      >
        <p>
          I have a tooltip
        </p>
      </div>
    </Tooltip>
  </Tooltip>
</WithTheme(Tooltip)>
`;
